lfsapi/client.go

```diff
@@ -15,6 +15,35 @@ import (
 
 var UserAgent = "git-lfs"
 
+const MediaType = "application/vnd.git-lfs+json; charset=utf-8"
+
+func (c *Client) NewRequest(method string, e Endpoint, suffix string, body interface{}) (*http.Request, error) {
+	req, err := http.NewRequest(method, joinURL(e.Url, suffix), nil)
+	if err != nil {
+		return req, err
+	}
+
+	req.Header.Set("Accept", MediaType)
+
+	if body != nil {
+		if merr := MarshalToRequest(req, body); merr != nil {
+			return req, merr
+		}
+		req.Header.Set("Content-Type", MediaType)
+	}
+
+	return req, err
+}
+
+const slash = "/"
+
+func joinURL(prefix, suffix string) string {
```

Comment:
It does extra stuff we don't need (calling `path.Clean()`) in our limited uses of `(*Client) NewRequest()` that we don't need. `suffix` is always a simple string like `"objects/batch"` or `"locks"`. Though I suppose anyone can import the package and call it with a weird `suffix`. Though I'm not sure why anyone would do that.